src/org/hammerhead226/sharkmacro/motionprofiles/ProfileHandler.java

```diff
@@ -182,84 +203,83 @@ public void manage() {
+		for (int i = 0; i < talons.length; i++) {
+			talons[i].set(ControlMode.MotionProfile, mode.value);
+		}
 	}
 
 	/**
-	 * Fill the Talons' top-level buffer with a given motion profile.
+	 * Fill the talons' top-level buffer with a given motion profile.
 	 * 
 	 */
 	private void fillTalonsWithMotionProfile() {
 
-		// maybe need two point objects?
-		TrajectoryPoint leftPoint = new TrajectoryPoint();
-		TrajectoryPoint rightPoint = new TrajectoryPoint();
-
 		if (profileIndex == 0) {
-			talons[0].clearMotionProfileTrajectories();
-			talons[0].configMotionProfileTrajectoryPeriod(TrajectoryDuration.Trajectory_Duration_0ms.value, 0);
-			talons[0].clearMotionProfileHasUnderrun(0);
-			talons[1].clearMotionProfileTrajectories();
-			talons[1].configMotionProfileTrajectoryPeriod(TrajectoryDuration.Trajectory_Duration_0ms.value, 0);
-			talons[1].clearMotionProfileHasUnderrun(0);
+			for (int i = 0; i < talons.length; i++) {
+				talons[i].clearMotionProfileTrajectories();
+				talons[i].configMotionProfileTrajectoryPeriod(TrajectoryDuration.Trajectory_Duration_0ms.value, 0);
+				talons[i].clearMotionProfileHasUnderrun(0);
+			}
 		}
 
-		talons[0].getMotionProfileStatus(leftStatus);
-		talons[1].getMotionProfileStatus(rightStatus);
+		updateMotionProfilesStatuses();
 
-		int numPointsToFill;
-		if (leftStatus.topBufferCnt > rightStatus.topBufferCnt) {
-			numPointsToFill = Constants.TALON_TOP_BUFFER_MAX_COUNT - leftStatus.topBufferCnt;
-		} else {
-			numPointsToFill = Constants.TALON_TOP_BUFFER_MAX_COUNT - rightStatus.topBufferCnt;
+		int maxFilled = statuses[0].topBufferCnt;
+		for (int i = 0; i < statuses.length; i++) {
+			if (statuses[i].topBufferCnt > maxFilled) {
+				maxFilled = statuses[i].topBufferCnt;
+			}
 		}
 
-		while (profileIndex < profiles[0].length && profileIndex < profiles[1].length && numPointsToFill > 0) {
-
-			leftPoint.position = profiles[0][profileIndex][0];
-			rightPoint.position = profiles[1][profileIndex][0];
-
-			leftPoint.velocity = profiles[0][profileIndex][1];
-			rightPoint.velocity = profiles[1][profileIndex][1];
-
-			leftPoint.headingDeg = 0;
-			rightPoint.headingDeg = 0;
-
-			leftPoint.timeDur = toTrajectoryDuration((int) profiles[0][profileIndex][2]);
-			rightPoint.timeDur = toTrajectoryDuration((int) profiles[1][profileIndex][2]);
-
-			leftPoint.profileSlotSelect0 = pidSlotIdxs[0];
-			rightPoint.profileSlotSelect0 = pidSlotIdxs[1];
-
-			leftPoint.profileSlotSelect1 = 0;
-			rightPoint.profileSlotSelect1 = 0;
-
-			leftPoint.zeroPos = false;
-			rightPoint.zeroPos = false;
-			if (profileIndex == 0) {
-				leftPoint.zeroPos = true;
-				rightPoint.zeroPos = true;
+		int numPointsToFill = Constants.TALON_TOP_BUFFER_MAX_COUNT - maxFilled;
+		
+		boolean notFinished = true;
+
+		while (notFinished && numPointsToFill > 0) {
+			for (int i = 0; i < trajPoints.length; i++) {
+				
+				if (profileIndex >= profiles[i].length) {
+					notFinished = false;
```

Comment:
I generally find these kinds of double negatives hard to read. `finished = true` would be easier.